app/Handlers/Utility/CurrencyHandler.php

```diff
@@ -0,0 +1,133 @@
+<?php
+
+namespace App\Handlers\Utility;
+
+use App\Slack\Event;
+use App\Slack\Message;
+use App\Handlers\Handler;
+use App\Loader\LoadsData;
+use App\Loader\JsonLoader;
+use Symfony\Component\Intl\Intl;
+use Illuminate\Support\Collection;
+
+final class CurrencyHandler extends Handler
+{
+    use LoadsData;
+
+    protected $dataFile = 'currency.json';
+
+    /**
+     * @param JsonLoader $loader
+     */
+    public function __construct(JsonLoader $loader)
+    {
+        $this->loader = $loader;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function canHandle(Event $event)
+    {
+        return
+            $event->isMessage() &&
+            ($event->isDirectMessage() || $event->mentions($this->eve->userId())) &&
+            $event->matches('/convert .+/i')
+        ;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function handle(Event $event)
+    {
+        $this->loadData();
+
+        $arguments = $this->arguments($event);
+        $content   = '`@eve convert 10 GPB to USD`';
+        $baseRate  = 1;
+
+        if ($arguments->count() == 4 && $this->validArguments($arguments)) {
+            $rates  = $this->data['rates'];
+            $result = '';
+
+            if (! array_key_exists($arguments[1], $rates) && ! array_key_exists($arguments[3], $rates)) {
+                return;
+            }
+
+            if (array_key_exists($arguments[1], $rates) && array_key_exists($arguments[3], $rates)) {
+                $result = number_format($arguments[0] * $rates[strtoupper($arguments[3])], 2);
+            }
+
+            if (array_key_exists($arguments[1], $rates) && ! array_key_exists($arguments[3], $rates)) {
+                if ($rates[strtoupper($arguments[1])] > $baseRate) {
```

Comment:
Nested ifs are never good. This needs to be refactored to make the logic clearer and follow some Object Calisthenics practices. Maybe you could extract some methods?